packages/shared/src/lib/core/activity/utils/evm/generateEvmTokenTransferActivityFromBlockscoutTokenTransfer.ts

```diff
@@ -64,6 +64,22 @@ export async function generateEvmTokenTransferActivityFromBlockscoutTokenTransfe
         return
     }
 
+    const contract: ISmartContractSubject | undefined = blockscoutTokenTransfer.to.is_contract
+        ? {
+              type: SubjectType.SmartContract,
+              address: blockscoutTokenTransfer.to.hash.toLowerCase(),
+              name: blockscoutTokenTransfer.to.name ?? '',
+              verified: blockscoutTokenTransfer.to.is_verified,
+          }
+        : blockscoutTokenTransfer.from.is_contract
+          ? {
+                type: SubjectType.SmartContract,
+                address: blockscoutTokenTransfer.to.hash.toLowerCase(),
+                name: blockscoutTokenTransfer.to.name ?? '',
+                verified: blockscoutTokenTransfer.to.is_verified,
+            }
+          : undefined
+
```

Comment:
Shouldn't the name be `blockScoutTokenTransfer.from` in case the contract sends this? Otherwise I would extract the condition and deduplicate setting the same object.

```suggestion
    const contract: ISmartContractSubject | undefined = ( blockscoutTokenTransfer.to.is_contract | blockscoutTokenTransfer.from.is_contract )
        ? {
              type: SubjectType.SmartContract,
              address: blockscoutTokenTransfer.to.hash.toLowerCase(),
              name: blockscoutTokenTransfer.to.name ?? '',
              verified: blockscoutTokenTransfer.to.is_verified,
          }
        : undefined

```